cmake/modules/njson.cmake

```diff
@@ -12,16 +12,34 @@
 # specific language governing permissions and limitations under the License.
 #
 
-if(USE_BUNDLED_NLOHMANN_JSON)
+#
+# njson (https://github.com/nlohmann/json)
+#
+
+option(USE_BUNDLED_NLOHMANN_JSON "Enable building of the bundled nlohmann-json" ${USE_BUNDLED_DEPS})
+
+if(nlohmann_json_INCLUDE_DIRS)
+    # we already have nlohmnann-json
+elseif(NOT USE_BUNDLED_NLOHMANN_JSON)
+    find_path(nlohmann_json_INCLUDE_DIRS NAMES nlohmann)
+
+    if(nlohmann_json_INCLUDE_DIRS)
+        message(STATUS "Found nlohmann-json: include: ${nlohmann_json_INCLUDE_DIRS}")
+    else()
+        message(FATAL_ERROR "Couldn't find system nlohmann-json")
+    endif()
+else()
+    set(nlohmann_json_INCLUDE_DIRS "${PROJECT_BINARY_DIR}/njson-prefix/include")
+
+    message(STATUS "Using bundled nlohmann-json in ${nlohmann_json_INCLUDE_DIRS}")
+
     ExternalProject_Add(njson
         URL "https://github.com/nlohmann/json/archive/v3.3.0.tar.gz"
         URL_HASH "SHA256=2fd1d207b4669a7843296c41d3b6ac5b23d00dec48dba507ba051d14564aa801"
         CMAKE_ARGS -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=${PROJECT_BINARY_DIR}/njson-prefix -DJSON_BuildTests=OFF -DBUILD_TESTING=OFF
     )
+endif()
 
-    set(nlohmann_json_INCLUDE_DIRS ${PROJECT_BINARY_DIR}/njson-prefix/include)
-else()
-    find_package(nlohmann_json CONFIG REQUIRED)
-    get_target_property(nlohmann_json_INCLUDE_DIRS nlohmann_json::nlohmann_json INTERFACE_INCLUDE_DIRECTORIES)
+if(NOT TARGET njson)
```

Comment:
I believe your change is ok. Can you just explain why we moved from `find_package`  to just `find_path`?

Reply:
I used `find_path` just to be consistent with the other cmake files.
But as @federico-sysdig pointed out using `find_package` is a better choice.
Note: `find_package` doesn't work with `cpp-httplib 0.10.3` (the version in the Ubuntu 22.04 repo) because the package doesn't provide `httplibConfig.cmake`.
However this should not be a problem since Falco requires `cpp-httplib 0.13.1` and this newer version does provide the cmake file required by `find_package`